connectors/es/sink.py

```diff
@@ -525,6 +527,89 @@ async def get_docs_incrementally(self, generator):
 
         await self.queue.put("END_DOCS")
 
+    async def get_access_control_docs(self, generator):
+        """Iterate on a generator of access control documents to fill a queue with bulk operations for the `Sink` to consume.
+
+        A document might be discarded if its timestamp has not changed.
+        """
+        logger.info("Starting access control doc lookups")
+        generator = self._decorate_with_metrics_span(generator)
+
+        self.sync_runs = True
+
+        existing_ids = {
+            doc_id: last_update_timestamp
+            async for (doc_id, last_update_timestamp) in self._get_existing_ids()
+        }
+
+        if logger.isEnabledFor(logging.DEBUG):
```

Comment:
just a comment, I am wondering if we could move this at the framework level, so all `debug` calls benefit from the lazy formating mechanism